Menu/Matcher/Voter/AdminVoter.php

```diff
@@ -49,9 +49,19 @@ public function matchItem(ItemInterface $item)
         $match = null;
         if ($admin instanceof AdminInterface
             && $admin->hasRoute('list') && $admin->hasAccess('list')
-            && $this->request && $this->request->get('_sonata_admin') == $admin->getCode()
+            && $this->request
         ) {
-            $match = true;
+            $requestAdmin = $admin->getConfigurationPool()->getAdminByAdminCode($this->request->get('_sonata_admin'));
+
+            if ($requestAdmin instanceof AdminInterface) {
+                if ($requestAdmin->getCode() === $admin->getCode()) {
+                    $match = true;
```

Comment:
I thought about it, then realized that every other menu voter uses the same useless `$match` variable. Should I do it in other menu voter too?